src/write.c

```diff
@@ -1558,6 +1558,12 @@ avifResult avifEncoderFinish(avifEncoder * encoder, avifRWData * output)
             avifRWStreamSetOffset(&s, prevOffset);
             avifRWStreamFinishBox(&s, stts);
 
+            // Check whether it is safe to cast width and height to uint16_t. The maximum width and
+            // height of an AV1 frame are 65536, which just exceeds uint16_t.
+            if ((imageMetadata->width > 65535) || (imageMetadata->height > 65535)) {
+                return AVIF_RESULT_INVALID_ARGUMENT;
+            }
```

Comment:
Joe: We need to cast `imageMetadata->width` and `imageMetadata->height` to `uint16_t` when writing out the VisualSampleEntry, at lines 1575-1576 below:

```
            avifRWStreamWriteU16(&s, (uint16_t)imageMetadata->width);  // unsigned int(16) width;
            avifRWStreamWriteU16(&s, (uint16_t)imageMetadata->height); // unsigned int(16) height;
```

Do you know where is the best place to check this?

Reply:
I discussed this with Joe. We will check this when we detect we will be writing an image sequence.